File: backend/app/schemas/article.py
from datetime import datetime
from pydantic import BaseModel, Field


class ArticleCreateRequest(BaseModel):
    title: str = Field(..., max_length=255)
    content: str


class ArticleResponse(BaseModel):
    id: int
    title: str
    content: str
    author_id: int | None = None
    author_name: str | None = None
    published: bool = False
    created_at: datetime

File: frontend/src/types/article.ts
export interface ArticleResponse {
  id: number;
  title: string;
  content: string;
  author_id?: number;
  author_name?: string;
  published: boolean;
  created_at?: string;
  updated_at?: string;
  image_url?: string;
}

export interface ArticleCreateRequest {
  title: string;
  content?: string;
  author_name?: string;
}

export interface ArticlePendingResponse extends ArticleResponse {}

export interface ArticleListResponse extends Array<ArticleResponse> {}
